Plugin silently fails when mypy `non_interactive` option is set

<!-- Hello! Thank you for submitting a bug to our repo. Our users and their bug reports are essential to our development process.
Before you submit an issue:
1. Please search for the bug in existing issues on the repo. Search using keywords or error messages and check both `open` and `closed` issues.
2. Try setting your log level to trace and look at the logs in more detail, as there may be more information there. If not, then these trace logs are part
of the bug report below! (see directions on how to set and access these logs in the `Logs` section below.)
Thank you!-->

## Diagnostic Data
- Python version (& distribution if applicable, e.g., Anaconda): 3.13.2
- Type of virtual environment used (e.g., conda, venv, virtualenv, etc.): venv, but probably any
- Operating system (and version): Fedora Linux 41, but again probably any
- Version of tool extension you are using: 2025.2.0

## Behaviour
### Expected Behavior
The plugin shows typing errors in the editor

![Image](https://github.com/user-attachments/assets/c7d6c65a-4bf6-42d1-a075-b841cc2def98)

### Actual Behavior
Even the most obvious typing errors do not show up in the editor

![Image](https://github.com/user-attachments/assets/45443af1-fe81-4fbd-87fb-de1f5b58a10e)

## Reproduction Steps:
Add a `setup.cfg` file to the project with this content:
```
[mypy]
install_types = True
non_interactive = True
```

Create a Python file with typing errors:
```Python
def broken() -> bool:
    an_int: int = 42
    a_string = "string"
    kaputt = an_int + a_string
    return kaputt
```
Behold the lack of errors shown in the editor.

## Logs:
First, without the `non_interactive` option, when everything works fine:
```
2025-04-22 17:14:01.400 [info] /bin/python -m mypy --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-codes --no-pretty --show-error-end /home/someone/DummyProject/mypy_kaputt.py
2025-04-22 17:14:01.401 [info] CWD Server: /home/someone/DummyProject
2025-04-22 17:14:01.677 [info] file:///home/someone/DummyProject/mypy_kaputt.py :
/home/someone/DummyProject/mypy_kaputt.py:4:23:4:30: error: Unsupported operand types for + ("int" and "str")  [operator]
/home/someone/DummyProject/mypy_kaputt.py:5:12:5:17: error: Incompatible return value type (got "int", expected "bool")  [return-value]

2025-04-22 17:14:01.677 [info] file:///home/someone/DummyProject/mypy_kaputt.py :
/home/someone/DummyProject/mypy_kaputt.py:4:23:4:30: error: Unsupported operand types for + ("int" and "str")  [operator]
/home/someone/DummyProject/mypy_kaputt.py:5:12:5:17: error: Incompatible return value type (got "int", expected "bool")  [return-value]
```

Now, with the `non_interactive` option set:
```
2025-04-22 17:11:54.005 [info] /bin/python -m mypy --no-color-output --no-error-summary --show-absolute-path --show-column-numbers --show-error-codes --no-pretty --show-error-end /home/someone/DummyProject/mypy_kaputt.py
2025-04-22 17:11:54.005 [info] CWD Server: /home/someone/DummyProject
2025-04-22 17:11:54.290 [info] /home/someone/DummyProject/mypy_kaputt.py:4:23:4:30: error: Unsupported operand types for + ("int" and "str")  [operator]
/home/someone/DummyProject/mypy_kaputt.py:5:12:5:17: error: Incompatible return value type (got "int", expected "bool")  [return-value]

2025-04-22 17:11:54.291 [info] file:///home/someone/DummyProject/mypy_kaputt.py :
```

Note that even though the typing errors are still in the log, the preceding `file:///` line is absent. It seems the plugin relies on this, even though the file path is also available from the error lines themselves.  
If presence of the `non_interactive` option causing the plugin to not report anything is considered expected behavior, then at least some warning should be shown about this. But ideally, the plugin should just work regardless of this setting.

--- a/bundled/tool/lsp_server.py
+++ b/bundled/tool/lsp_server.py
@@ -235,10 +235,17 @@ def _linting_helper(document: workspace.Document) -> None:
             extra_args += ["--show-error-end"]
 
         result = _run_tool_on_document(document, extra_args=extra_args)
-        if result and result.stdout:
-            log_to_output(f"{document.uri} :\r\n{result.stdout}")
+        # Some mypy modes (e.g., non_interactive) emit diagnostics on stderr.
+        # Prefer parsing combined output so we don't miss errors when stdout is empty.
+        if result and (result.stdout or result.stderr):
+            combined_output = "\n".join(
+                [s for s in [result.stdout or "", result.stderr or ""] if s]
+            )
+            # Keep existing stdout logging for consistency; stderr is logged separately above.
+            if result.stdout:
+                log_to_output(f"{document.uri} :\r\n{result.stdout}")
             parse_results = _parse_output_using_regex(
-                result.stdout, settings["severity"]
+                combined_output, settings["severity"]
             )
             reportingScope = settings["reportingScope"]
             for file_path, diagnostics in parse_results.items():

